components/src/core/InfoListItem/InfoListItem.tsx

```diff
@@ -193,20 +193,24 @@ const InfoListItemRender: React.ForwardRefRenderFunction<unknown, InfoListItemPr
         }
     }, [icon, avatar, hidePadding, combine]);
 
-    const getRightComponent = useCallback((): JSX.Element | undefined => {
-        if (rightComponent) {
-            return <RightComponent className={combine('rightComponent')}>{rightComponent}</RightComponent>;
-        } else if (chevron) {
-            return (
-                <InfoListItemChevron
-                    chevronColor={chevronColor}
-                    color={'inherit'}
-                    role={'button'}
-                    className={combine('chevron')}
-                />
-            );
-        }
-    }, [rightComponent, chevron, combine]);
+    const getRightComponent = useCallback(
+        (): JSX.Element | undefined => (
+            <>
+                {rightComponent && (
+                    <RightComponent className={combine('rightComponent')}>{rightComponent}</RightComponent>
+                )}
+                {chevron && (
+                    <InfoListItemChevron
+                        chevronColor={chevronColor}
+                        color={'inherit'}
+                        role={'button'}
+                        className={combine('chevron')}
+                    />
+                )}
+            </>
+        ),
+        [rightComponent, chevron, combine]
```

Comment:
I was a bit surprised that the `rightComponent` prop is not necessary for this dependency list while dev doc playground still works.

Reply:
I'm not sure I understand this comment...rightComponent _is_ in the dependency array.

Reply:
my bad, meant to type `chevronColor`

Reply:
yeah, chevronColor should be in there. If it's working anyway, we're probably just getting lucky and the entire component is re-rendering based on some other prop change or something like that.